plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/JvmExamplePlugin.java

```diff
@@ -111,5 +117,16 @@ public Settings additionalSettings() {
     public void onModule(RepositoriesModule repositoriesModule) {
     }
 
-
+    /**
+     * Module decalaring some example configuration and a _cat action that uses
+     * it.
+     */
+    public static class ConfiguredExampleModule extends AbstractModule {
+        @Override
+        protected void configure() {
+          bind(ExamplePluginConfiguration.class).asEagerSingleton();
+          Multibinder<AbstractCatAction> catActionMultibinder = Multibinder.newSetBinder(binder(), AbstractCatAction.class);
```

Comment:
No, there are only two ways for sets/maps, because of how multibinder works. I think it is more confusing to have some things bound using a binder, and others with registration/settings. Classes that ES controls should be registered, and set. If there was a way to just not allow multiple multibinders to work I would say we should do that, but I don't think such a thing exists.